### PR TITLE
Cura 11851 warning message for extruder2

### PR DESCRIPTION
**Pull request by saumyaj3**
# Description

Add core compatibility check for Factor4 machine

A new function has been added to the MachineManager class to check for core compatibility in the Factor4 machine. This function ensures the selected core is compatible for use with the Factor4 machine and if not, it displays a warning icon with an appropriate message.

CURA-11851

### CODE REVIEW

File: resources/qml/Menus/ConfigurationMenu/CustomConfiguration.qml [hunk 1]
```diff
@@ -345,8 +346,8 @@ Item
                         anchors.left: warningImage.right
                         anchors.leftMargin: UM.Theme.getSize("default_margin").width
                         width: selectors.controlWidth - warningImage.width - UM.Theme.getSize("default_margin").width
-                        text: catalog.i18nc("@label", "Use glue for better adhesion with this material combination.")
-                        visible: CuraSDKVersion == "dev" ? false : warnings.buildplateCompatibilityError || warnings.buildplateCompatibilityWarning
+                        text:warnings.coreCompatibilityWarning? catalog.i18nc("@label", "Combination not recommended. Load BB core to slot 1 (left) for better reliability."): catalog.i18nc("@label", "Use glue for better adhesion with this material combination.")
+                        visible: CuraSDKVersion == "dev" ? false : warnings.buildplateCompatibilityError || warnings.buildplateCompatibilityWarning|| warnings.coreCompatibilityWarning
                         wrapMode: Text.WordWrap
                     }
```
Comment (casperlamboo) [inline resources/qml/Menus/ConfigurationMenu/CustomConfiguration.qml:352]: Why not add a different UM.Label here with a different visibility check

Something like this i mean
```python
                    UM.Label
                    {
                        id: materialCompatibilityLabel
                        anchors.left: warningImage.right
                        anchors.leftMargin: UM.Theme.getSize("default_margin").width
                        width: selectors.controlWidth - warningImage.width - UM.Theme.getSize("default_margin").width
                        text: catalog.i18nc("@label", "Use glue for better adhesion with this material combination.")
                        visible: CuraSDKVersion == "dev" ? false : warnings.buildplateCompatibilityError || warnings.buildplateCompatibilityWarning
                        wrapMode: Text.WordWrap
                    }
                    
                    UM.Label
                    {
                        id: materialCompatibilityLabel
                        anchors.left: warningImage.right
                        anchors.leftMargin: UM.Theme.getSize("default_margin").width
                        width: selectors.controlWidth - warningImage.width - UM.Theme.getSize("default_margin").width
                        text: catalog.i18nc("@label", "Combination not recommended. Load BB core to slot 1 (left) for better reliability.")
                        visible: warnings.coreCompatibilityWarning
                        wrapMode: Text.WordWrap
                    }
```

File: cura/Settings/MachineManager.py [hunk 2]
```diff
@@ -847,6 +847,24 @@ def variantBuildplateUsable(self) -> bool:
...
         return result
 
+    @pyqtProperty(bool, notify = activeMaterialChanged)
+    def variantCoreUsableForFactor4(self) -> bool:
+        """The selected core is usable if it is in second extruder of Factor4
+        """
+        result = True
+        if not self._global_container_stack:
+            return result
+        if self.activeMachine.id != "UltiMaker Factor 4":
```
Comment (casperlamboo) [inline cura/Settings/MachineManager.py:857]: we can't use this check, creating two factor4's results in the following machine id: `UltiMaker Factor 4 #2`

File: cura/Settings/MachineManager.py [hunk 3]
```diff
@@ -847,6 +847,24 @@ def variantBuildplateUsable(self) -> bool:
...
+            return result
+
+        for extruder_container in self._global_container_stack.extruderList:
+            if( extruder_container.id.startswith("ultimaker_factor4_extruder_right")):
```
Comment (casperlamboo) [inline cura/Settings/MachineManager.py:861]: also you should use the definition id
```suggestion
            if extruder_container.definition.id.startswith("ultimaker_factor4_extruder_right"):
```

File: cura/Settings/MachineManager.py (hunk 2 above)
Comment (saumyaj3) [reply cura/Settings/MachineManager.py:857]: Oh no.

File: resources/qml/Menus/ConfigurationMenu/CustomConfiguration.qml (hunk 1 above)
Comment (saumyaj3) [reply resources/qml/Menus/ConfigurationMenu/CustomConfiguration.qml:363]: done 6542d9ff1334d176d75ed9b2a8e7aba633f88040

File: cura/Settings/MachineManager.py (hunk 2 above)
Comment (saumyaj3) [reply cura/Settings/MachineManager.py]: fixed here 6542d9ff1334d176d75ed9b2a8e7aba633f88040

File: cura/Settings/MachineManager.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (saumyaj3) [reply cura/Settings/MachineManager.py]: done 6542d9ff1334d176d75ed9b2a8e7aba633f88040

[the remark this answers is not in the document]
Comment (saumyaj3) [reply cura/Settings/MachineManager.py]: done 6542d9ff1334d176d75ed9b2a8e7aba633f88040

File: cura/Settings/MachineManager.py [hunk 4]
```diff
@@ -847,6 +847,24 @@ def variantBuildplateUsable(self) -> bool:
...
         return result
 
+    @pyqtProperty(bool, notify = currentConfigurationChanged)
+    def variantCoreUsableForFactor4(self) -> bool:
+        """The selected core is usable if it is in second extruder of Factor4
+        """
+        result = True
+        if not self._global_container_stack:
+            return result
+        if not self.activeMachine.id.startswith("UltiMaker Factor 4"):
```
Comment (casperlamboo) [inline cura/Settings/MachineManager.py:857]: this must really be this
```suggestion
        if self.activeMachine.definition.id != "ultimaker_factor4":
```
we want to check for equality against the definition id, this is constant no matter what machine instance you have

**Pull request merged**